client/src/app/plan.service.ts

```diff
@@ -513,4 +526,67 @@ export class PlanService {
       };
     }
   }
+
+  /**
+   * Creates an observable for submitting the edited degree plan to the server
+   * @param plan The transformed degree plan
+   */
+  submitPlan(plan: any): Observable<any> {
+    const semesters: Array<any> = this.convertSemestersForSubmission(
+      plan.years
+    );
+
+    return this.http.post(
+      `${this.userService.uri}/plan/${plan.id}`,
+      { semesters },
+      this.userService.getHttpHeaders()
+    );
+  }
+
+  /**
+   * Reverses the data transformation made by the component into the format the backend API can read
+   * @param years The years array
+   */
+  private convertSemestersForSubmission(years: Array<any>): Array<any> {
```

Comment:
Reverses data transformations made by the component so that the backend can process the new plan